Utils/trigger_test_upload_flow.sh

```diff
@@ -1,60 +1,473 @@
 #!/usr/bin/env bash
 
-if [ "$#" -lt "1" ]; then
-  echo "Usage:
-  $0 -ct <token>
-
-  -ct, --ci-token             The ci token.
-  [-b, --branch]              The branch name. Default is the current branch.
-  [-gb, --bucket]             The name of the bucket to upload the packs to. Default is marketplace-dist-dev.
-  [-f, --force]               Whether to trigger the force upload flow.
-  [-p, --packs]               CSV list of pack IDs. Mandatory when the --force flag is on.
-  [-ch, --slack-channel]      A slack channel to send notifications to. Default is dmst-bucket-upload.
-  [-g, --gitlab]              Flag indicating to trigger the flow in GitLab.
-  "
+# This script creates new branch with changes that will test the upload flow with given sdk and content branches.
+# Note: This script creates new remote branch, please delete the branch once the pipeline finished.
+
+# fail
+# show fail message and quit
+# :param $1: message
+# :param $2: Skip deleting branch
+function fail {
+  echo "$1"
+  if [ -z $2 ]; then
+    git checkout "${content_branch_name}"
+    git branch -D "${new_content_branch}" # delete local branch
+  fi
   exit 1
-fi
+}
+
+# check_arguments
+# Check if the given arguments are valid
+function check_arguments {
+  echo " Running - check_arguments"
+
+  if [ -z "$content_branch_name" ]; then
+    content_branch_name=$(git branch --show-current)
+  fi
+
+  if [ -z "$gitlab_token" ] && [ -z "$circle_token" ]; then
+    fail "At least one token [-gt, --gitlab-ci-token] or [-ct, --circle-ci-token] is required." "skip"
+  fi
+
+  if [ -n "$force" ] && [ -z "$packs" ]; then
+    fail "You must provide a csv list of packs to force upload." "skip"
+  fi
+
+}
+
+# copy_pack
+# Copies a pack and changing the name in the files to the new name.
+# :param $1: pack name
+# :param $2: suffix for new pack name
+# :param $3: possible names for renaming array
+function create_new_pack {
+  echo " Running - create_new_pack"
+
+  local pack_name=$1
+  local new_pack_suffix=$2
+  shift
+  shift
+  local names_array=("$@")
+
+  local new_pack_name="${pack_name}${new_pack_suffix}"
+  local original_path=$(pwd)
+  local pack_path="${CONTENT_PATH}/Packs/${pack_name}"
+  local new_pack_path="${CONTENT_PATH}/Packs/${new_pack_name}"
+
+  cp -R "${pack_path}" "${new_pack_path}" || fail
+  cd "${new_pack_path}" || fail
+
+ rename_files_and_folders "$pack_name" "$new_pack_name"
+
+  for original_name in "${names_array[@]}"; do
+    new_name="${original_name}${new_pack_suffix}"
+    rename_files_and_folders "$original_name" "$new_name"
+  done
```

Comment:
what is this doing?

Reply:
Renaming all the files and folder with original name, and the occurrence inside the files.

Making sure that the copied pack will not contain any used id.